src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +47,43 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        StringBuilder sb = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + "\n");
+        LocalDate d1 = convertToLocalDate(dateFrom);
+        LocalDate d2 = convertToLocalDate(dateTo);
+        if (d1.isAfter(d2)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        data = filterByDate(data, dateFrom, dateTo);
+        for (String name : names) {
+            int count = 0;
+            for (String line : data) {
+                String[] arr = line.split("[ ]");
+                if (arr[1].equals(name)) {
+                    count = count + Integer.parseInt(arr[2]) * Integer.parseInt(arr[3]);
+                }
+            }
+            sb.append(name).append(" - ").append(count).append("\n");
+        }
+        return sb.toString().substring(0, sb.toString().length() - 1);
+    }
+
+    private static LocalDate convertToLocalDate(String s) {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return LocalDate.parse(s, dtf);
+    }
+
+    private static String[] filterByDate(String[] data, String dateFrom, String dateTo) {
+        List<String> list = new ArrayList<>();
+        for (String s : data) {
+            LocalDate d1 = convertToLocalDate(s.substring(0, 10));
```

Comment:
d1, d2, d3 is not meaningful names. Please rename. Variable name should be informative